client/src/Components/Notifications/index.js

```diff
@@ -0,0 +1,73 @@
+/* eslint-disable no-unused-vars */
+import { useEffect, useState } from 'react';
+import { Link } from 'react-router-dom';
+
+const Notifications = ({ username }) => {
+	const [notifications, setNotifications] = useState(
+		localStorage.getItem('notifications')
+			? JSON.parse(localStorage.getItem('notifications'))
+			: []
+	);
+
+	// for opening and EventSource with the server
+	useEffect(() => {
+		if (username) {
+			const eventSource = new EventSource(
+				`http://localhost:3001/api/v1/users/${username}/notifications`,
+				{ withCredentials: true }
+			);
```

Comment:
Since you open the connection and set a listener, I assume that you need to clear it after this component is unmounted.